### DISCUSSION TITLE
How to make it work with Artillery Sidewinder X2

### DISCUSSION THREAD

**Discussion by sakomoto987** [OrcaSlicer/OrcaSlicer#934] [Q&A] [answered]
Hello,

I tried to operate the software with my Sidewinder X2 printer, without success. Whether with the original G-Code or the custom that is used in my usual slicer (S3D). 
The machine constantly reaches the borders of the device, does not put itself in Z0 and seems to be reboot before an impression.

Here is what I usually use:

```
G28 ; home

;G29 ; Autoleveling au lancement
;M500 ; savegarde EEPROM

M420 S1 Z10 ; rappel du dernier Autoleveling pour gagner du temps avec stop a 10 mm

M117 ; Purge extrudeur
G92 E0 ; reinitialisation extrudeur
M201 X800 Y800 ; acceleration max
M205 X10.0 Z0.30 ; reduction inertie
G1 Z0.0 F3000
G1 X0.1 Y20 Z0.3 F5000.0 ; deplacement position intiale
G1 X0.1 Y200.0 Z0.3 F750.0 E15 ; premiere ligne
G1 X0.4 Y200.0 Z0.3 F5000.0
G1 X0.4 Y20 Z0.3 F750.0 E30 ; deuxieme ligne
; G1 E27 F1800 ; retraction du filament
G92 E0 ;  reinitialisation extrudeur
G1 Z1.0 F3000 ; on bouge un peu la tete pour eviter les frottements
;Goldorak Go
```

**Comment by fr3ak2402** [comment]
Hi @sakomoto987, I will have a look at the problem later and try to solve it. Can you maybe test the G-code afterwards?

**Reply by fr3ak2402** [reply]
Start-Code:
`; Initial setups\nG90 ; use absolute coordinates\nM83 ; extruder relative mode\nM900 K0.12 ; K factor\nM900 W[line_width] H[layer_height] D[filament_diameter]\nM200 D0 ; disable volumetric e\nM220 S100 ; reset speed factor to 100%\nM221 S100 ; reset extrusion rate to 100%\n\n; Set the heating\nM190 S[hot_plate_temp_initial_layer] ; wait for bed to heat up\nM104 S[nozzle_temperature_initial_layer] ; start nozzle heating but don't wait\n\n; Home\nG1 Z3 F3000 ; move z up little to prevent scratching of surface\nG28 ; home all axes\nG1 X3 Y3 F5000 ; move to corner of the bed to avoid ooze over centre\n\n; Wait for final heating\nM109 S[nozzle_temperature_initial_layer] ; wait for the nozzle to heat up\nM190 S[hot_plate_temp_initial_layer] ; wait for the bed to heat up\n\n;Auto bed Leveling\n@BEDLEVELVISUALIZER\nG29 ; ABL T\nM420 S1 Z3 ; reload and fade mesh bed leveling until it reach 3mm Z\n\n; Return to prime position, Prime line routine\nG92 E0 ; Reset Extruder\nG1 Z3 F3000 ; move z up little to prevent scratching of surface\nG1 X10 Y.5 Z0.25 F5000.0 ; Move to start position\nG1 X100 Y.5 Z0.25 F1500.0 E15 ; Draw the first line\nG1 X100 Y.2 Z0.25 F5000.0 ; Move to side a little\nG1 X10 Y.2 Z0.25 F1500.0 E30 ; Draw the second line\nG92 E0 ; Reset Extruder\nM221 S{if layer_height<0.075}100{else}95{endif}`

End-Code:
`G4 ; wait\nG92 E0 ; prepare to retract\nG1 E-0.5 F3000; retract to avoid stringing\n\n; Anti-stringing end wiggle\nG91 ; use relative coordinates\nG1 X1 Y1 F1200\n\n; Raise nozzle and present bed\n{if layer_z < printable_height}G1 Z{z_offset+min(layer_z+120, printable_height)}{endif} ; Move print head up\nG90 ; use absolute coordinates\n\n; Reset print setting overrides\nM200 D0 ; disable volumetric e\nM220 S100 ; reset speed factor to 100%\nM221 S100 ; reset extrusion rate to 100%\n\n; Shut down printer\nM106 S0 ; turn-off fan\nM104 S0 ; turn-off hotend\nM140 S0 ; turn-off bed\nM150 P0 ; turn off led\nM85 S0 ; deactivate idle timeout\nM84 ; disable motors\n`

I have written you a start code and an end code for the Artillery Sidewinder X2, this is based on the PrusaSlicer. The auto-leveling feature of your X2 should also be executed.

Maybe it runs better with these codes and also fixes your problem.

**Reply by sakomoto987** [reply]
Hi @fr3ak2402 , thanks for your attempt.

I replaced every _\n_ with a _return line_.
Unfortunately, the printer don't launch ABL procedure, and also starts from it's relative position (same for Z) than from the home position.
Do you have X2 Sidewinder? Does it work fine at your end?

**Reply by fr3ak2402** [reply]
I don't own a Sidewinder X2 myself, but know my way around such settings.
I think I found the error why the printer crashes and does not start printing. Do you see an error message on the printer display?

I think it is not the G-code but a parameter of the profile.

Please change the following values in the machine settings, as follows comes to the machine settings
![Screenshot 2023-05-03 180940](https://user-images.githubusercontent.com/96239814/235975960-115f2ffd-c3ee-4577-a98a-274a3abe14bf.png)

Then switch to the Acceleration setting tab and adjust the following as shown in the picture.

![Screenshot 2023-05-03 180820](https://user-images.githubusercontent.com/96239814/235976276-4932c6b6-dffe-484e-9926-b4ccea98f97c.png)


The problem is with the maximum speed of Z, this is set to 0.4 and so the X2 has a problem. Because this causes that no movement can be done on the Z axis.

**Reply by fr3ak2402** [reply]
After you have adjusted the values, you can slice an object and test it. This should solve your problem, if there is still a problem, I will take a closer look at the profile.

If this solves your problem, then I will also publish this, so that in the next release of the OrcaSlicer is changed for all

**Comment by sakomoto987** [comment]
Let me detail. The printer doesn't crash, before printing it doesn't find the home.

Where it is at the beginning of a print, it's the home / origin for the X2.

I have no error message displayed.

I kept the g-code you shared and I also change the printer setting like in your capture.

**Reply by fr3ak2402** [reply]
I have heard from one person that the Sidewinder X2 works for this person and am now getting the complete profile of the printer. Can you then try the profile once?

**Reply by sakomoto987** [reply]
Yeah I tried the yesterday gcode (the long one) + changing the inputs like you screenshot.

I don't know if it can cause that, but I set the nozzle parameter to 0.6 and also made my trial on a température calibration

**Reply by fr3ak2402** [reply]
I'll get the profile from the Artillery Sidewinder X2, which works for the person in OrcaSlicer.  Then the problem should be solved for you and everyone else.

**Reply by sakomoto987** [reply]
OK I'll wait the official release

**Reply by fr3ak2402** [reply]
I read around a bit and found a post where there was a hint for the start code:

![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/96239814/236782739-d3d4a5b8-d330-4cb9-8a6d-d0add9327f7d.png)

So I have adapted your G-code a little bit:

G28 ; home

M420 S1 Z2 ; rappel du dernier Autoleveling pour gagner du temps avec stop a 2 mm

;G29 ; Autoleveling au lancement
;M500 ; savegarde EEPROM

M117 ; Purge extrudeur
G92 E0 ; reinitialisation extrudeur
M201 X800 Y800 ; acceleration max
M205 X10.0 Z0.30 ; reduction inertie
G1 Z0.0 F3000
G1 X0.1 Y20 Z0.3 F5000.0 ; deplacement position intiale
G1 X0.1 Y200.0 Z0.3 F750.0 E15 ; premiere ligne
G1 X0.4 Y200.0 Z0.3 F5000.0
G1 X0.4 Y20 Z0.3 F750.0 E30 ; deuxieme ligne
; G1 E27 F1800 ; retraction du filament
G92 E0 ;  reinitialisation extrudeur
G1 Z1.0 F3000 ; on bouge un peu la tete pour eviter les frottements
;Goldorak Go

I hope it fixes your problem. Your printer also works completely in all other slices?

**Reply by sakomoto987** [reply]
Thanks for your investigation, but I'm pretty sure it will not improve the situation.

Z2 instead of Z10 means that the bed leveling mapping compensation is stop at the Z level indicated.

Yeah this code works on S3D.

**Reply by fr3ak2402** [reply]
@sakomoto987 

I have completely rebuilt the profile of the Artillery Sidewinder X2 and also had it tested.

The test was successful and an OctoPrint connection was also possible with the X2. As well as the auto leveling of the printer.

The new profile will be available in the next nightly build or in the next official release.

**Reply by sakomoto987** [reply]
Hello @fr3ak2402  Thanks! I'm happy testing it

**Comment by sakomoto987** [comment, marked as answer]
Official release 1.6.3 fix ASX2 printing. Thanks @fr3ak2402
